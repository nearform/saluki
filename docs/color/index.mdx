import { Color, Container, Box, Text } from './'

# Color

---

## Example

<Color>Hello</Color>

```js
const Text = styled.h1`
  ${color.white}
  ${border.solid}
  ${border.blue}
  ${background.color.black}
`
```

## Palette

<Container>
  <Box color="black">
    <Text light={true}>black</Text>
  </Box>
  <Box color="white">
    <Text>white</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterGray">
    <Text>lighterGray</Text>
  </Box>
  <Box color="lightGray">
    <Text>lightGray</Text>
  </Box>
  <Box color="gray">
    <Text light={true}>gray</Text>
  </Box>
  <Box color="darkGray">
    <Text light={true}>darkGray</Text>
  </Box>
  <Box color="darkerGray">
    <Text light={true}>darkerGray</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterRed">
    <Text>lighterRed</Text>
  </Box>
  <Box color="lightRed">
    <Text>lightRed</Text>
  </Box>
  <Box color="red">
    <Text light={true}>red</Text>
  </Box>
  <Box color="darkRed">
    <Text light={true}>darkRed</Text>
  </Box>
  <Box color="darkerRed">
    <Text light={true}>dakerRed</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterBlue">
    <Text>lighterBlue</Text>
  </Box>
  <Box color="lightBlue">
    <Text>lightBlue</Text>
  </Box>
  <Box color="blue">
    <Text light={true}>blue</Text>
  </Box>
  <Box color="darkBlue">
    <Text light={true}>darkBlue</Text>
  </Box>
  <Box color="darkerBlue">
    <Text light={true}>darkerBlue</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterYellow">
    <Text>lighterYellow</Text>
  </Box>
  <Box color="lightYellow">
    <Text>lightYellow</Text>
  </Box>
  <Box color="yellow">
    <Text light={true}>yellow</Text>
  </Box>
  <Box color="darkYellow">
    <Text light={true}>darkYellow</Text>
  </Box>
  <Box color="darkerYellow">
    <Text light={true}>darkerYellow</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterGreen">
    <Text>lighterGreen</Text>
  </Box>
  <Box color="lightGreen">
    <Text>lightGreen</Text>
  </Box>
  <Box color="green">
    <Text light={true}>green</Text>
  </Box>
  <Box color="darkGreen">
    <Text light={true}>darkGreen</Text>
  </Box>
  <Box color="darkerGreen">
    <Text light={true}>darkerGreen</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterOrange">
    <Text>lighterOrange</Text>
  </Box>
  <Box color="lightOrange">
    <Text>lightOrange</Text>
  </Box>
  <Box color="orange">
    <Text light={true}>orange</Text>
  </Box>
  <Box color="darkOrange">
    <Text light={true}>darkOrange</Text>
  </Box>
  <Box color="darkerOrange">
    <Text light={true}>darkerOrange</Text>
  </Box>
</Container>

<Container>
  <Box color="lighterPurple">
    <Text>lighterPurple</Text>
  </Box>
  <Box color="lightPurple">
    <Text>lightPurple</Text>
  </Box>
  <Box color="purple">
    <Text light={true}>purple</Text>
  </Box>
  <Box color="darkPurple">
    <Text light={true}>darkPurple</Text>
  </Box>
  <Box color="darkerPurple">
    <Text light={true}>darkerPurple</Text>
  </Box>
</Container>

## Rules

```js
color.black
color.white

color.lighterGray
color.lightGray
color.gray
color.darkGray
color.darkerGray

color.lighterRed
color.lightRed
color.red
color.darkRed
color.darkerRed

color.lighterBlue
color.lightBlue
color.blue
color.darkBlue
color.darkerBlue

color.lighterYellow
color.lightYellow
color.yellow
color.darkYellow
color.darkerYellow

color.lighterGreen
color.lightGreen
color.green
color.darkGreen
color.darkerGreen

color.lighterOrange
color.lightOrange
color.orange
color.darkOrange
color.darkerOrange

color.lighterPurple
color.lightPurple
color.purple
color.darkPurple
color.darkerPurple
```

## Configuration

Configurable via;

```
color
```

## Defaults

```js
{
  black: 'black',
  white: 'white',
  lighterGray: '#d5d9dc',
  lightGray: '#9ea6ae',
  gray: '#495057',
  darkGray: '#3a4045',
  darkerGray: '#23262a',
  lighterRed: '#f9b8b8',
  lightRed: '#f25959',
  red: '#f03e3e',
  darkRed: '#a60d0d',
  darkerRed: '#470606',
  lighterBlue: '#bbdaf6',
  lightBlue: '#61a9ea',
  blue: '#1c7ed6',
  darkBlue: '#155d9e',
  darkerBlue: '#092844',
  lighterYellow: '#ffe4b3',
  lightYellow: '#ffc04d',
  yellow: '#f59f00',
  darkYellow: '#b37400',
  darkerYellow: '#4d3200',
  lighterGreen: '#c5edcc',
  lightGreen: '#77d587',
  green: '#37b24d',
  darkGreen: '#2a883b',
  darkerGreen: '#123a19',
  lighterOrange: '#fdd2b5',
  lightOrange: '#fa9551',
  orange: '#f76707',
  darkOrange: '#ae4805',
  darkerOrange: '#4a1f02',
  lighterPurple: '#e6c3ee',
  lightPurple: '#c474d8',
  purple: '#ae3ec9',
  darkPurple: '#78278b',
  darkerPurple: '#33113c'
}
```
