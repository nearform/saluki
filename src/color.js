export const color = {
  red: 'red'
}
