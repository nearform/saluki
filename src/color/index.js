export const defaultRules = {
  black: 'black',
  white: 'white',
  lighterGray: '#d5d9dc',
  lightGray: '#9ea6ae',
  gray: '#495057',
  darkGray: '#3a4045',
  darkerGray: '#23262a',
  lighterRed: '#f9b8b8',
  lightRed: '#f25959',
  red: '#f03e3e',
  darkRed: '#a60d0d',
  darkerRed: '#470606',
  lighterBlue: '#bbdaf6',
  lightBlue: '#61a9ea',
  blue: '#1c7ed6',
  darkBlue: '#155d9e',
  darkerBlue: '#092844',
  lighterYellow: '#ffe4b3',
  lightYellow: '#ffc04d',
  yellow: '#f59f00',
  darkYellow: '#b37400',
  darkerYellow: '#4d3200',
  lighterGreen: '#c5edcc',
  lightGreen: '#77d587',
  green: '#37b24d',
  darkGreen: '#2a883b',
  darkerGreen: '#123a19',
  lighterOrange: '#fdd2b5',
  lightOrange: '#fa9551',
  orange: '#f76707',
  darkOrange: '#ae4805',
  darkerOrange: '#4a1f02',
  lighterPurple: '#e6c3ee',
  lightPurple: '#c474d8',
  purple: '#ae3ec9',
  darkPurple: '#78278b',
  darkerPurple: '#33113c'
}

const setupColor = colors => {
  let color = {}
  for (const [key, value] of Object.entries(colors)) {
    color[key] = {
      color: value
    }
  }
  return color
}

export default setupColor
