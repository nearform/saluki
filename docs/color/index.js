import styled from 'styled-components/macro'
import { background, border, color } from '../../src'

export const Color = styled.h1`
  ${color.white}
  ${border.solid}
  ${border.blue}
  ${background.black}
`
