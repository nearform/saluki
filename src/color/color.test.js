import color, { defaultRules } from './'

describe('color', () => {
  describe('when no custom config is provided', () => {
    it('should return the default config', () => {
      expect(color(defaultRules)).toMatchSnapshot()
    })
  })

  describe('when custom config is provided', () => {
    describe('when the custom config overrides a value', () => {
      it('should replace the default value', () => {
        const combinedRules = {
          ...defaultRules,
          ...{
            red: '#ff0000'
          }
        }
        expect(color(combinedRules)).toMatchSnapshot()
      })
    })

    describe('when the custom config provides a new value', () => {
      it('should add the value and not amend any existing values', () => {
        const combinedRules = {
          ...defaultRules,
          ...{
            darkestRed: '#540000'
          }
        }
        expect(color(combinedRules)).toMatchSnapshot()
      })
    })
  })
})
