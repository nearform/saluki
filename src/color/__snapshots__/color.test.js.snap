// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color when custom config is provided when the custom config overrides a value should replace the default value 1`] = `
Object {
  "black": Object {
    "color": "black",
  },
  "blue": Object {
    "color": "#1c7ed6",
  },
  "darkBlue": Object {
    "color": "#155d9e",
  },
  "darkGray": Object {
    "color": "#3a4045",
  },
  "darkGreen": Object {
    "color": "#2a883b",
  },
  "darkOrange": Object {
    "color": "#ae4805",
  },
  "darkPurple": Object {
    "color": "#78278b",
  },
  "darkRed": Object {
    "color": "#a60d0d",
  },
  "darkYellow": Object {
    "color": "#b37400",
  },
  "darkerBlue": Object {
    "color": "#092844",
  },
  "darkerGray": Object {
    "color": "#23262a",
  },
  "darkerGreen": Object {
    "color": "#123a19",
  },
  "darkerOrange": Object {
    "color": "#4a1f02",
  },
  "darkerPurple": Object {
    "color": "#33113c",
  },
  "darkerRed": Object {
    "color": "#470606",
  },
  "darkerYellow": Object {
    "color": "#4d3200",
  },
  "gray": Object {
    "color": "#495057",
  },
  "green": Object {
    "color": "#37b24d",
  },
  "lightBlue": Object {
    "color": "#61a9ea",
  },
  "lightGray": Object {
    "color": "#9ea6ae",
  },
  "lightGreen": Object {
    "color": "#77d587",
  },
  "lightOrange": Object {
    "color": "#fa9551",
  },
  "lightPurple": Object {
    "color": "#c474d8",
  },
  "lightRed": Object {
    "color": "#f25959",
  },
  "lightYellow": Object {
    "color": "#ffc04d",
  },
  "lighterBlue": Object {
    "color": "#bbdaf6",
  },
  "lighterGray": Object {
    "color": "#d5d9dc",
  },
  "lighterGreen": Object {
    "color": "#c5edcc",
  },
  "lighterOrange": Object {
    "color": "#fdd2b5",
  },
  "lighterPurple": Object {
    "color": "#e6c3ee",
  },
  "lighterRed": Object {
    "color": "#f9b8b8",
  },
  "lighterYellow": Object {
    "color": "#ffe4b3",
  },
  "orange": Object {
    "color": "#f76707",
  },
  "purple": Object {
    "color": "#ae3ec9",
  },
  "red": Object {
    "color": "#ff0000",
  },
  "white": Object {
    "color": "white",
  },
  "yellow": Object {
    "color": "#f59f00",
  },
}
`;

exports[`color when custom config is provided when the custom config provides a new value should add the value and not amend any existing values 1`] = `
Object {
  "black": Object {
    "color": "black",
  },
  "blue": Object {
    "color": "#1c7ed6",
  },
  "darkBlue": Object {
    "color": "#155d9e",
  },
  "darkGray": Object {
    "color": "#3a4045",
  },
  "darkGreen": Object {
    "color": "#2a883b",
  },
  "darkOrange": Object {
    "color": "#ae4805",
  },
  "darkPurple": Object {
    "color": "#78278b",
  },
  "darkRed": Object {
    "color": "#a60d0d",
  },
  "darkYellow": Object {
    "color": "#b37400",
  },
  "darkerBlue": Object {
    "color": "#092844",
  },
  "darkerGray": Object {
    "color": "#23262a",
  },
  "darkerGreen": Object {
    "color": "#123a19",
  },
  "darkerOrange": Object {
    "color": "#4a1f02",
  },
  "darkerPurple": Object {
    "color": "#33113c",
  },
  "darkerRed": Object {
    "color": "#470606",
  },
  "darkerYellow": Object {
    "color": "#4d3200",
  },
  "darkestRed": Object {
    "color": "#540000",
  },
  "gray": Object {
    "color": "#495057",
  },
  "green": Object {
    "color": "#37b24d",
  },
  "lightBlue": Object {
    "color": "#61a9ea",
  },
  "lightGray": Object {
    "color": "#9ea6ae",
  },
  "lightGreen": Object {
    "color": "#77d587",
  },
  "lightOrange": Object {
    "color": "#fa9551",
  },
  "lightPurple": Object {
    "color": "#c474d8",
  },
  "lightRed": Object {
    "color": "#f25959",
  },
  "lightYellow": Object {
    "color": "#ffc04d",
  },
  "lighterBlue": Object {
    "color": "#bbdaf6",
  },
  "lighterGray": Object {
    "color": "#d5d9dc",
  },
  "lighterGreen": Object {
    "color": "#c5edcc",
  },
  "lighterOrange": Object {
    "color": "#fdd2b5",
  },
  "lighterPurple": Object {
    "color": "#e6c3ee",
  },
  "lighterRed": Object {
    "color": "#f9b8b8",
  },
  "lighterYellow": Object {
    "color": "#ffe4b3",
  },
  "orange": Object {
    "color": "#f76707",
  },
  "purple": Object {
    "color": "#ae3ec9",
  },
  "red": Object {
    "color": "#f03e3e",
  },
  "white": Object {
    "color": "white",
  },
  "yellow": Object {
    "color": "#f59f00",
  },
}
`;

exports[`color when no custom config is provided should return the default config 1`] = `
Object {
  "black": Object {
    "color": "black",
  },
  "blue": Object {
    "color": "#1c7ed6",
  },
  "darkBlue": Object {
    "color": "#155d9e",
  },
  "darkGray": Object {
    "color": "#3a4045",
  },
  "darkGreen": Object {
    "color": "#2a883b",
  },
  "darkOrange": Object {
    "color": "#ae4805",
  },
  "darkPurple": Object {
    "color": "#78278b",
  },
  "darkRed": Object {
    "color": "#a60d0d",
  },
  "darkYellow": Object {
    "color": "#b37400",
  },
  "darkerBlue": Object {
    "color": "#092844",
  },
  "darkerGray": Object {
    "color": "#23262a",
  },
  "darkerGreen": Object {
    "color": "#123a19",
  },
  "darkerOrange": Object {
    "color": "#4a1f02",
  },
  "darkerPurple": Object {
    "color": "#33113c",
  },
  "darkerRed": Object {
    "color": "#470606",
  },
  "darkerYellow": Object {
    "color": "#4d3200",
  },
  "gray": Object {
    "color": "#495057",
  },
  "green": Object {
    "color": "#37b24d",
  },
  "lightBlue": Object {
    "color": "#61a9ea",
  },
  "lightGray": Object {
    "color": "#9ea6ae",
  },
  "lightGreen": Object {
    "color": "#77d587",
  },
  "lightOrange": Object {
    "color": "#fa9551",
  },
  "lightPurple": Object {
    "color": "#c474d8",
  },
  "lightRed": Object {
    "color": "#f25959",
  },
  "lightYellow": Object {
    "color": "#ffc04d",
  },
  "lighterBlue": Object {
    "color": "#bbdaf6",
  },
  "lighterGray": Object {
    "color": "#d5d9dc",
  },
  "lighterGreen": Object {
    "color": "#c5edcc",
  },
  "lighterOrange": Object {
    "color": "#fdd2b5",
  },
  "lighterPurple": Object {
    "color": "#e6c3ee",
  },
  "lighterRed": Object {
    "color": "#f9b8b8",
  },
  "lighterYellow": Object {
    "color": "#ffe4b3",
  },
  "orange": Object {
    "color": "#f76707",
  },
  "purple": Object {
    "color": "#ae3ec9",
  },
  "red": Object {
    "color": "#f03e3e",
  },
  "white": Object {
    "color": "white",
  },
  "yellow": Object {
    "color": "#f59f00",
  },
}
`;
